### PR TITLE
Improved code for selection of measurement start

### PR DESCRIPTION
**Pull request by GabrielePalazzo**
Fixed the start of the measurement pulse and improved code readability.

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

File: src/qibolab/instruments/zhinst.py [hunk 1]
```diff
@@ -892,39 +892,18 @@ def measure_relax(self, exp, qubits, relaxation_time, acquisition_type):
...
-                else self.play_after_set(self.sequence_qibo.qd_pulses, "drive")
-            )
-        if len(self.sequence_qibo.cf_pulses) != 0 and len(self.sequence_qibo.qd_pulses) != 0:
-            play_after = (
-                self.play_after_set(self.sequence_qibo.cf_pulses, "bias_coupler")
-                if self.sequence_qibo.cf_pulses.finish > self.sequence_qibo.qd_pulses.finish
-                else self.play_after_set(self.sequence_qibo.qd_pulses, "drive")
-            )
+        # TODO: if we use sweepers, the code might not behave as expected
+        # i.e.: self.sequence_qibo may not contain ZhSweeper for flux
```
Comment (Jacfomg) [inline src/qibolab/instruments/zhinst.py:896]:
```suggestion
        # TODO: if we use duration sweepers, the code might not behave as expected
        # i.e.: self.sequence_qibo will contain the a pulse or sweeper with a static duration that may screw the comparison
```

File: src/qibolab/instruments/zhinst.py [hunk 2]
```diff
@@ -948,6 +927,9 @@ def measure_relax(self, exp, qubits, relaxation_time, acquisition_type):
                 for pulse, q, iq_angle in zip(pulses, qubits, iq_angles):
                     pulse.zhpulse.uid += str(i)
 
+                    # TODO: if the measure sequence starts after the last pulse, add a delay
+                    # if the delay is too large (i.e. 5us),
```
Comment (Jacfomg) [inline src/qibolab/instruments/zhinst.py:931]: Could you check the maximun before it crashes ?

**Review by Jacfomg — APPROVED** [review]
Apart from the documentation for the errors the code seems to work as before.

File: src/qibolab/instruments/zhinst.py (hunk 2 above)
Comment (GabrielePalazzo) [reply src/qibolab/instruments/zhinst.py:931]: Sure.

**Comment by Jacfomg** [comment]
If you removed the maximun delay comment can you document what we talked ?

**Review by stavros11 — APPROVED** [review]

File: src/qibolab/instruments/zhinst.py [hunk 3]
```diff
@@ -505,7 +505,9 @@ def register_couplerflux_line(self, coupler):
 
     def run_exp(self):
         """Compilation settings, compilation step, execution step and data retrival"""
+        # self.experiment.save("saved_exp")  # saving experiment
         self.exp = self.session.compile(self.experiment, compiler_settings=COMPILER_SETTINGS)
+        # self.exp.save("saved_exp")  # saving compiled experiment
         self.results = self.session.run(self.exp)
```
Comment (AleCandido) [inline src/qibolab/instruments/zhinst.py:511]: Before merging, please drop dead code.

Comment (Jacfomg) [reply src/qibolab/instruments/zhinst.py:511]: i asked him to get those lines there to document how to save the experiments in the case Zurich supports asks for them. I would like to have them somewhere.

Comment (AleCandido) [reply src/qibolab/instruments/zhinst.py:511]: Then I would suggest the docstring :)

**Pull request merged**